src/java/voldemort/store/readonly/ReadOnlyStorageConfiguration.java

```diff
@@ -48,7 +49,18 @@ public ReadOnlyStorageConfiguration(VoldemortConfig config) {
         this.nodeId = config.getNodeId();
         this.deleteBackupMs = config.getReadOnlyDeleteBackupMs();
         this.maxValueBufferAllocationSize = config.getReadOnlyMaxValueBufferAllocationSize();
-        this.voldConfig = config;
+
+        FailedFetchLock failedFetchLockInstance;
+        if (config.getHighAvailabilityStateAutoCleanUp()) {
+            try {
+                failedFetchLockInstance = FailedFetchLock.getLock(config, null);
```

Comment:
Sure, I could create a new ```FailedFetchLock``` and then close it every time we need one, rather than hanging on to it.